File: myblogsfrontend/src/app/payload/PostEditViewResponse.ts
export class PostEditViewResponse {

    constructor(

        // post id
        public postid: number,

        public creator: string,

        // title
        public title: string,

        // category id
        public categoryid: number,
        
        public categoryName: string,

        // category id + category name
        public category: string,

        // post content
        public content: string,

        // the post is created on this datetime
        public createDatetime: Date,

        // last update date time
        public lastUpdateDatetime: Date,


    ) {

    }

}

File: myblogsbackend/src/main/java/com/ez/myblogsbackend/payload/PostEditViewResponse.java
package com.ez.myblogsbackend.payload;

import java.util.Date;

public interface PostEditViewResponse {

    // "postid" column
    Long getPostid();

    // "creator id + creator email" column
    String getCreator();

    // "title" column
    String getTitle();

    // "content" column
    String getContent();

    // "createDatetime" column
    Date getCreateDatetime();

    // "lastUpdateDatetime" column
    Date getLastUpdateDatetime();

    // "categoryid" column
    long getCategoryid();

    public String getCategoryName();

}
